File: memory-cache/csharp/tests/MemoryCache.Tests/FixedClock.cs
namespace MemoryCache.Tests;

public class FixedClock : IClock
{
    private DateTime _now;

    public FixedClock(DateTime now) => _now = now;

    public DateTime Now() => _now;

    public void AdvanceTo(DateTime when) => _now = when;

    public void Advance(TimeSpan delta) => _now = _now.Add(delta);
}

File: zombie-survivor/typescript/tests/FixedClock.ts
import type { Clock } from '../src/Clock.js';

export class FixedClock implements Clock {
  private _now: Date;

  constructor(now: Date) {
    this._now = now;
  }

  now(): Date { return this._now; }

  advanceTo(date: Date): void { this._now = date; }
}
